util/stop_watch.h

```diff
@@ -33,13 +37,35 @@ class StopWatch {
         *elapsed_ += env_->NowMicros() - start_time_;
       }
     }
+    if (elapsed_ && delay_enabled_) {
+      *elapsed_ -= total_delay_;
+    }
     if (stats_enabled_) {
       statistics_->measureTime(hist_type_,
           (elapsed_ != nullptr) ? *elapsed_ :
                                   (env_->NowMicros() - start_time_));
     }
   }
 
+  void DelayStart() {
+    // if delay_start_time_ is not 0, it means we are already tracking delay,
+    // so delay_start_time_ should not be overwritten
+    if (delay_enabled_ && delay_start_time_ == 0) {
```

Comment:
i think we should also check `elapsed_ != nullptr` before taking any measurements